src/print-results.ts

```diff
@@ -251,3 +269,37 @@ function sumCounts(results: ESLint.LintResult[]): TodoFormatterCounts {
 function pluralize(word: string, count: number): string {
   return count === 1 ? word : `${word}s`;
 }
+
+function filterTodos(
+  results: ESLint.LintResult[],
+  options: TodoPrintOptions
+): ESLint.LintResult[] {
+  return results.reduce((acc, result) => {
+    const messages = result.messages as TodoResultMessage[];
+    if (messages.length === 0) {
+      return acc;
+    }
```

Comment:
The return statement at the end of this function will handle the case where there's no messages, so no real need to early return here.

```suggestion
```

Reply:
This is the same as the comment below about filtering out all todos. With this case, the result might look something like:

```
[
  { filePath: '/foo', messages: [{ ... }, { ... }]}
]
```

Without this case, it might look like:

```
[
  { filePath: '/foo', messages: [{ ... }, { ... }]},
  { filePath: '/bar', messages: []},
  { filePath: '/baz', messages: []},
]
```